src/types/tx.ts

```diff
@@ -0,0 +1,75 @@
+import { EncodeObject } from "@cosmjs/proto-signing";
+import { StdFee } from "@cosmjs/stargate";
+import { CreateTxOptions, Msg } from "@terra-money/terra.js";
+import { ConnectedWallet } from "@terra-money/wallet-provider";
+import { Chain } from "./aws";
+import { EVMTx, SimulContractTx, SimulSendNativeTx } from "./evm";
+import { TagPayload } from "./third-party/redux";
+
+// //////////// SEND TX ////////////
+export type CosmosTx = {
+  msgs: EncodeObject[];
+  fee: StdFee;
+};
+
+export type EstimatedTx =
+  | { type: "cosmos"; val: CosmosTx }
+  | {
+      type: "terra";
+      val: CreateTxOptions;
+      wallet: ConnectedWallet /**future client/provider will be included in wallet */;
+    }
+  | { type: "evm"; val: EVMTx };
+
+export type SubmittedTx = { hash: string; chainID: string };
+
+export type TxLoading = { loading: string };
+export type TxError = { error: string; tx?: SubmittedTx };
+export type TxSuccess = SubmittedTx & {
```

Comment:
nit: could've kept `TxSuccess` consistent with `TxError` and have it have a nested field `tx: SubmittedTx`.

Reply:
having them different is easier for type guards,
```
//success
{ hash, chainID }

//error
{error tx? } 
```
 no need for them to be consistent since they are two types that manifest different scenario